adobe_vipm/flows/utils.py

```diff
@@ -1076,3 +1078,64 @@ def get_address(data):
         "addressLine2": data.get("addressLine2", ""),
         "postCode": data.get("postalCode", ""),
     }
+
+def is_line_item_active_subscription(subscriptions, line):
+    adobe_item = get_item_by_partial_sku(
+        subscriptions["items"], line["item"]["externalIds"]["vendor"]
+    )
+    return adobe_item["status"] == STATUS_SUBSCRIPTION_ACTIVE
+
+
+def has_valid_returnable_quantity(line, returnable_orders):
+    delta = line["oldQuantity"] - line["quantity"]
+    total_quantity_returnable = sum(roi.quantity for roi in returnable_orders)
+    if delta != total_quantity_returnable:
+        return False
+    return True
+
+
+def validate_subscription_and_returnable_orders(
+    adobe_client,
+    context,
+    line,
+    sku,
+    return_orders=None
+):
+    """
+    Validates if the subscription is active and has valid returnable orders.
+
+    Args:
+        adobe_client: The Adobe client instance
+        context: The context object
+        line: The order line to validate
+        sku: The SKU to validate
+        return_orders: Optional return orders to pass to get_returnable_orders_by_sku
+
+    Returns:
+        tuple: (is_valid, returnable_orders)
+        - is_valid: True if validation passed, False otherwise
+        - returnable_orders: List of returnable orders if validation passed, None otherwise
+    """
```

Comment:
I think we should add a little bit of logging here, espacially for all the returning branches, otherwise to understand what happend in production without it would be painfull :-(